### ISSUE TITLE
What's the best method for using jQuery to expand/collapse select elements?

### ISSUE THREAD

**Issue by ayumihamasaki2019** [jquery/jquery#4486]
Hi

I'm having some problems trying to get jQuery to work with `select` / `option` elements, so I have decided to ask the experts here.

I have the following code example:

```html
<div class="form-group">

    <select class="form-control valid" name="example" aria-label="We require you to select an option" required aria-required="true" role="listbox" aria-haspopup="listbox" aria-expanded="false">
        <option value="">Select an option</option>
        <option value="red">Red</option>
        <option value="blue">Blue</option>
        <option value="green">Green</option>
    </select>

</div>
```

**I have no problem with mouse clicking, my issues is with keyboard navigation.**

I'm trying to set the attribute `aria-expanded="false"` and `aria-expanded="true"` as per the select element being expanded and collapsed.

For example:

### Collapsed

![image](https://user-images.githubusercontent.com/55155131/65153773-3c7bcb80-da22-11e9-9c0d-6fca0a4137a8.png)

```html
<select aria-expanded="false">
```

### Expanded

![image](https://user-images.githubusercontent.com/55155131/65153840-5cab8a80-da22-11e9-866e-43213fd6224c.png)

```html
<select aria-expanded="true">
```

To make things more complicated than they need to be Google Chrome allows the following keyboard actions to open and close the select element:

```
- Enter 
- Alt + Arrow Up
- Alt + Arrow Down
```

For Firefox they allow:

```
- Alt + Arrow Up
- Alt + Arrow Down
```

Note: `Enter` key does nothing in Firefox.

Basically how does jQuery detect and know when a `select` element is **expanded** or **collasped** and what is the best method to use?

Because **Chrome let's you expand** the `select` element and **_Firefox and IE etc do not_**. So I need a more complex logic to double check the browser has expanded the `select` element or not before adding/updating the `aria-expanded` attribute.

Thanks.

**Comment by timmywil** [comment]
Thanks for opening an issue. This repo is for tracking bugs and features. You can find support on stackoverflow.com or on [jQuery's gitter](https://gitter.im/jquery/jquery).
